public/js/admin/users.js

```diff
@@ -86,22 +86,44 @@ var createUsers = {
     },
     validateRolesAndPassword: function () {
         var pass = true;
-        //validate password
+        // Adding password constraint!
+        var passwordErrors = "";
+        var password = $.trim(document.forms["createForm"]["password"].value);
+        if(password.length < 6){
+            passwordErrors += "password is less than 6 characters ";
+            //createUsers.elements.password.next(".errors").text("password is less than 6 characters\n");
+            pass = false;
+        }
+        var re = /[A-Z]/;
+        if(!re.test(password)){
+            passwordErrors += "password must have at least one upper case ";
+            //createUsers.elements.password.next(".errors").text("password must have at least one upper case\n");
+            pass = false;
+        }
+        re = /[0-9]/;
         if ($.trim(document.forms["createForm"]["password"].value) === "") {
-            createUsers.elements.password.next(".errors").text("please assign this user a password");
+            passwordErrors = "please assign this user a password";
             pass = false;
-        } else {
+        }
+        else if(!re.test(password)) {
+            passwordErrors +=  "password does not have a number ";
+            // createUsers.elements.password.next(".errors").text("password does not have a number\n");
+            pass = false;
+        }
+
+        if(pass == false)
```

Comment:
AJ,

You can use a strict comparison here. JavaScript uses both '==' and '==='. '===' won't match anything other than false. For example, this if statement will end up being true in the following scenarios while using the '==' operator:

pass = 0
pass = '0'